src/typescript-service.ts

```diff
@@ -338,6 +338,847 @@ export default class TypeScriptService {
 		});
 	}
 
+	getDocumentSymbol(params: DocumentSymbolParams): Promise<SymbolInformation[]> {
+		const uri = params.textDocument.uri;
+		return this.ensureFilesForHoverAndDefinition(uri).then(() => {
+			const fileName = util.uri2path(uri);
+			const config = this.projectManager.getConfiguration(uri);
+			return config.prepare(this.connection).then(() => {
+				this.projectManager.syncConfiguration(config, this.connection);
+				const sourceFile = config.program.getSourceFile(fileName);
+				const tree = config.service.getNavigationTree(fileName);
+				const limit = 100000;
```

Comment:
This is meant to mean "effectively unlimited" and is only used once, so I inlined it instead of giving it a name.